torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1185,13 +1185,15 @@ void ProcessGroupNCCL::shutdown() {
 }
 
 ProcessGroupNCCL::~ProcessGroupNCCL() {
+  LOG(INFO) << logPrefix() << "ProcessGroupNCCL destructor entered.";
   terminateProcessGroup_.store(true);
   workMetaListCV_.notify_one();
 
 #ifdef ENABLE_NCCL_ERROR_CHECKING
   if (ncclCommWatchdogThread_.joinable()) {
     ncclCommWatchdogThread_.join();
   }
+  LOG(INFO) << logPrefix() << "ProcessGroupNCCL watchdog thread joined.";
```

Comment:
Just curious, what case will these logs help debug?

Reply:
So we want to ensure that the job hangs at the destructor since we got a lot of reports recently.

Reply:
hmm. link?

is heartbeat monitor killing in these cases? or is the job exiting normally but then hanging at dtor?